app/assets/javascripts/admin/addon/components/themes-grid-placeholder.gjs

```diff
@@ -28,59 +28,325 @@ export default class ThemesGridPlaceholder extends Component {
     }
   }
 
+  get safeThemeColors() {
+    let colors = this.themeColors;
+
+    return {
+      primary: htmlSafe(colors.primary),
+      secondary: htmlSafe(colors.secondary),
+      tertiary: htmlSafe(colors.tertiary),
+      quaternary: htmlSafe(colors.quaternary),
+      highlight: htmlSafe(colors.highlight),
+      danger: htmlSafe(colors.danger),
+      success: htmlSafe(colors.success),
+      love: htmlSafe(colors.love),
+    };
+  }
+
+  get randomVariant() {
+    return (this._variant ||= Math.floor(Math.random() * 4) + 1);
+  }
+
+  get gradientId() {
+    return `bgGradient-${this.args.theme.id}-${this.randomVariant}`;
+  }
+
+  get placeholderStyle1() {
+    return this.randomVariant === 1;
+  }
+
+  get placeholderStyle2() {
+    return this.randomVariant === 2;
+  }
+
+  get placeholderStyle3() {
+    return this.randomVariant === 3;
+  }
+
+  get placeholderStyle4() {
+    return this.randomVariant === 4;
+  }
```

Comment:
Instead of defining these getters I'd recommend using `import { eq } from "truth-helpers";` and then inlining these in the template, e.g:

```hbs
{{#if (eq this.randomVariant 4)}}
```

same effect and performance, just fewer lines and all the needed context is within the template